### PR TITLE
[translation] AAD support

### PR DESCRIPTION
**Pull request by kristapratico**
Resolves https://github.com/Azure/azure-sdk-for-python/issues/18883

### CODE REVIEW

File: sdk/translation/azure-ai-translation-document/README.md [hunk 1]
```diff
@@ -86,6 +86,35 @@ credential = AzureKeyCredential("<api_key>")
...
+Note that regional endpoints do not support AAD authentication. Create a [custom subdomain][custom_subdomain]
+name for your resource in order to use this type of authentication.
+
+To use the [DefaultAzureCredential][default_azure_credential] type shown below, or other credential types provided
+with the Azure SDK, please install the `azure-identity` package:
+
+```pip install azure-identity```
+
+You will also need to [register a new AAD application and grant access][register_aad_app] to
+Form Recognizer by assigning the `"Cognitive Services User"` role to your service principal.
```
Comment (maririos) [inline sdk/translation/azure-ai-translation-document/README.md:102]: reference to form recognizer.

On a side note, is it true the service also needs this role?

File: sdk/translation/azure-ai-translation-document/README.md [hunk 2]
```diff
@@ -86,6 +86,35 @@ credential = AzureKeyCredential("<api_key>")
...
+Once completed, set the values of the client ID, tenant ID, and client secret of the AAD application as environment variables:
+`AZURE_CLIENT_ID`, `AZURE_TENANT_ID`, `AZURE_CLIENT_SECRET`.
+
+```python
+from azure.identity import DefaultAzureCredential
+from azure.ai.translation.document import DocumentTranslationClient
+credential = DefaultAzureCredential()
+
+document_translation_client = DocumentTranslationClient(
+    endpoint="https://<my-custom-subdomain>.cognitiveservices.azure.com/",
```
Comment (maririos) [inline sdk/translation/azure-ai-translation-document/README.md:113]: in the above examples, when talking about endpoint the reference is to resource name and not the subdomain. 
to maintain consistency, consider using the same everywhere.
i.e. `endpoint = "https://<resource-name>.cognitiveservices.azure.com/"`

File: sdk/translation/azure-ai-translation-document/README.md [hunk 3]
```diff
@@ -86,6 +86,35 @@ credential = AzureKeyCredential("<api_key>")
 document_translation_client = DocumentTranslationClient(endpoint, credential)
 ```
 
+#### Create the client with an Azure Active Directory credential
```
Comment (maririos) [inline sdk/translation/azure-ai-translation-document/README.md:89]: out of curiosity, do you know if there are any service docs about supporting AAD?

**Review by maririos — APPROVED** [review]
Concept and documentation LGTM.
Not sure on the Python specifics :)

File: sdk/translation/azure-ai-translation-document/README.md (hunk 1 above)
Comment (kristapratico) [reply sdk/translation/azure-ai-translation-document/README.md:102]: My understanding is that AAD is managed the same for all cognitive services. The docs here [apply](https://docs.microsoft.com/en-us/azure/cognitive-services/authentication?tabs=powershell#authenticate-with-azure-active-directory) and recommend this role type.

File: sdk/translation/azure-ai-translation-document/README.md (hunk 2 above)
Comment (kristapratico) [reply sdk/translation/azure-ai-translation-document/README.md:113]: Good point, making that change!

**Review by maririos — APPROVED** [review]

**Pull request merged**